Sources/SocksCore/Conversions.swift

```diff
@@ -20,9 +20,30 @@ extension Array {
 }
 
 extension String {
-    
     public func toBytes() -> [UInt8] {
         return Array(self.utf8)
     }
 }
 
+// Sendable and Receivable implementations
+
+extension String: Sendable {
+    public func serialize() -> [UInt8] {
+        return self.toBytes()
+    }
+}
+
+extension String: Receivable {
+    static func deserialize(reader: (maxBytes: Int) throws -> [UInt8]) throws -> String {
```

Comment:
Nevermind, I see now that it's -> Self, so it is a plausible scenario where the closure would be sustained beyond scope. Good catch 👍